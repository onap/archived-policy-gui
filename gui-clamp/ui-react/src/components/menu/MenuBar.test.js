/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * Modifications Copyright (C) 2022 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import MenuBar from './MenuBar';
import DropdownItem from "react-bootstrap/DropdownItem";

describe('Verify MenuBar', () => {

  it('Test the render method', () => {
    const component = shallow(<MenuBar/>)

    expect(component).toMatchSnapshot();
  });

  it('Update loopName', () => {
    const component = shallow(<MenuBar/>)
    component.setProps({ loopName: "newLoop" });
    expect(component.state('disabled')).toBe(false);
  });

  it('Default loopName', () => {
    const component = shallow(<MenuBar/>)
    component.setProps({ loopName: "Empty (NO loop loaded yet)" });
    expect(component.state('disabled')).toBe(true);
  });

  it('Find 20 DropdownItem', () => {
    const component = shallow(<MenuBar />)
    expect(component.find('DropdownItem').length).toEqual(20);
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(0).dive();
    expect(component.find({ tile: 'POLICY Framework' }));
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(1).dive();
    expect(component.find({ tile: 'CLAMP Options' }));
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(2).dive();
    expect(component.find({ tile: 'LOOP Instance' }));
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(3).dive();
    expect(component.find({ tile: 'LOOP Operations' }));
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(4).dive();
    expect(component.find({ tile: 'TOSCA Automation Composition' }));
  });

  it('Finds POLICY Framework Menu', () => {
    const component = shallow(<MenuBar />).childAt(5).dive();
    expect(component.find({ tile: 'Help' }));
  });

  it('Finds StyledNavLink', () => {
    const component = shallow(<MenuBar />);
    expect(component.find('Styled(NavLink)').length).toEqual(2);
  });

  it('Finds StyledNavDropdown', () => {
    const component = shallow(<MenuBar />);
    expect(component.find('Styled(NavDropdown)').length).toEqual(6);
  });
});
