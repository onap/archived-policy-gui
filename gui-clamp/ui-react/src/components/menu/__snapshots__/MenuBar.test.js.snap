// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify MenuBar Test the render method 1`] = `
<NavbarCollapse>
  <Styled(NavDropdown)
    title="Tosca"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/readToscaTemplate"
    >
      View Tosca Template
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="POLICY Framework"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/viewAllPolicies"
    >
      View All Policies
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="CLAMP Options"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/manageDictionaries"
    >
      Tosca Metadata Dictionaries
    </DropdownItem>
    <DropdownDivider
      role="separator"
    />
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/viewLoopTemplatesModal"
    >
      View All Loop Templates
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="LOOP Instance"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/createLoop"
    >
      Create
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/openLoop"
    >
      Open
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/closeLoop"
    >
      Close
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/modifyLoop"
    >
      Modify
    </DropdownItem>
    <DropdownDivider
      role="separator"
    />
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/loopProperties"
    >
      Properties
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/refreshStatus"
    >
      Refresh Status
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="LOOP Operations"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/submit"
    >
      Create and deploy to Policy Framework (SUBMIT)
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/stop"
    >
      Undeploy from Policy Framework (STOP)
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/restart"
    >
      ReDeploy to Policy Framework (RESTART)
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/delete"
    >
      Delete loop instance (DELETE)
    </DropdownItem>
    <DropdownDivider
      role="separator"
    />
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/deploy"
    >
      Deploy to DCAE (DEPLOY)
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={true}
      to="/undeploy"
    >
      UnDeploy to DCAE (UNDEPLOY)
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="TOSCA Control Loop"
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/monitoring"
    >
      Monitoring Control Loop
    </DropdownItem>
  </Styled(NavDropdown)>
  <Styled(NavDropdown)
    title="Help"
  >
    <Styled(NavLink)
      href="https://wiki.onap.org/"
      target="_blank"
    >
      Wiki
    </Styled(NavLink)>
    <Styled(NavLink)
      href="mailto:onap-discuss@lists.onap.org?subject=CLAMP&body=Please send us suggestions or feature enhancements or defect. If possible, please send us the steps to replicate any defect."
    >
      Contact Us
    </Styled(NavLink)>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: normal;
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  text-align: inherit;
  white-space: nowrap;
  border: 0;

  :hover {
    text-decoration: none;
    background-color: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      to="/userInfo"
    >
      User Info
    </DropdownItem>
  </Styled(NavDropdown)>
</NavbarCollapse>
`;
